connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -900,6 +995,8 @@ private void startWork() {
             callables.add(getConnectorStartingCallable(connectorName));
         }
 
+        runningAssignment.tasks().removeAll(tasksToRestart);
+        tasksToRestart.clear();
```

Comment:
Minor: it would help to have a comment here to explain why the tasks to restart are removed from the running tasks.

Reply:
Done